### PR TITLE
try delay the check version action while app start

### PR DESCRIPTION
**Pull request by shun2wang**
make a test branch and build please.

### CODE REVIEW

File: Desktop/gui/jaspversionchecker.cpp [hunk 1]
```diff
@@ -46,7 +46,7 @@ void JASPVersionChecker::downloadKnownIssues()
 void JASPVersionChecker::downloadVersionFinished()
 {
 	QString version			= _networkReply->readAll().trimmed(),
-			downloadfile	= "https://jasp-stats.org/download/";
+			downloadfile	= "http://jasp-stats.org/download/";  //Do not do use https here because on windows some PC has ssl verify problems.
```
Comment (JorisGoosen) [inline Desktop/gui/jaspversionchecker.cpp]: This is only used when one clicks the download button.
Id rather not remove https here, as that could allow mitm to lead people to the wrong (possibly spoofed) downloadpage

Comment (shun2wang) [reply Desktop/gui/jaspversionchecker.cpp]: Ah, yes!

**Comment by JorisGoosen** [comment]
Ive pushed them to jasp-stats and turned on nightlies, but I think I broke windows so maybe only those for macos will actually build

**Comment by JorisGoosen** [comment]
@RensDofferhoff  will build a version now (thanks!)

File: Desktop/main.cpp [hunk 2]
```diff
@@ -434,10 +434,12 @@ int main(int argc, char *argv[])
 			if(safeGraphics)
 			{
 				std::cout << "Setting special options for software rendering (aka safe graphics)." << std::endl;
-				QCoreApplication::setAttribute(Qt::AA_UseSoftwareOpenGL);
+				//QCoreApplication::setAttribute(Qt::AA_UseSoftwareOpenGL);
+				QCoreApplication::setAttribute(Qt::AA_UseDesktopOpenGL);
+				QQuickWindow::setGraphicsApi(QSGRendererInterface::OpenGL);
```
Comment (shun2wang) [inline Desktop/main.cpp:440]: I imagine if we could try different rendering backends by passing startup parameters in order to debug and solve the problem of qtWebengine init/crashing on certain graphics cards and rendering backends. any ideas? @JorisGoosen

**Comment by shun2wang** [comment]
Sorry I missed include "qquickwindow.h" so maybe we should re-trigger it again.

**Comment by RensDofferhoff** [comment]
@shun2wang I will first make a build of development branch and then this one. Could you rebase on development or should I?

**Comment by shun2wang** [comment]
ah yes please do it. thanks

File: Desktop/main.cpp [hunk 3]
```diff
@@ -434,10 +435,12 @@ int main(int argc, char *argv[])
...
 				args.push_back("--disable-gpu");
 				char dst[] = "LIBGL_ALWAYS_SOFTWARE=1";
-				putenv(dst);
+				_putenv(dst);
```
Comment (JorisGoosen) [inline Desktop/main.cpp]: this breaks unix builds

File: Desktop/utilities/imgschemehandler.cpp [hunk 4]
```diff
@@ -12,6 +13,8 @@ using namespace Modules;
 ImgSchemeHandler::ImgSchemeHandler(QObject *parent)
 	: QWebEngineUrlSchemeHandler{parent}
 {
+    std::cout << "ImgSchemeHandler for QtWebengine" << std::endl;
```
Comment (shun2wang) [inline Desktop/utilities/imgschemehandler.cpp:16]: now the application stoped here to start and continue initialize...any ideas to go debug here? @JorisGoosen 

or build a release with debug symbols version?

Comment (shun2wang) [reply Desktop/utilities/imgschemehandler.cpp:16]: in logs now we can see:
```
Setting special options for software rendering (aka safe graphics).
Making new argument list for Application startup: JASP.exe --logToFile --disable-gpu
Starting JASP 0.19 from commit 0e4d561737223e223956950a6c9f6d5a5e1b3f6d and branch lockfiletesting
QtWebEngineQuick initialized
Application initialized
PlotSchemeHandler for QtWebengine
```

File: Desktop/main.cpp (hunk 2 above)
Comment (JorisGoosen) [reply Desktop/main.cpp:440]: Reading https://www.qt.io/blog/new-graphics-integration-in-qt-webengine-6.6-and-even-6.5.1 it looks like you can set it via an environment variable as `QSG_RHI_BACKEND=opengl` and thus do not need the changes you made (for testing anyway).

Not sure if there are other settings that make sense. In any case, we just switched to 6.7 and I guess theyve now removed the old opengl backend. So the new nightly should be behaving differently even in opengl mode if I understand correctly.

With the environment variable it is possible to test if unsetting it makes a difference in 0.18.3 (it was built with 6.5.3 I think)

File: Desktop/utilities/imgschemehandler.cpp [hunk 5]
```diff
@@ -12,6 +13,8 @@ using namespace Modules;
...
+
 	QQuickWebEngineProfile::defaultProfile()->installUrlSchemeHandler("img", this);
 }
```
Comment (JorisGoosen) [inline Desktop/utilities/imgschemehandler.cpp:19]:
```suggestion
	std::cout << "scheme installed" << std::endl;
}
```

Comment (JorisGoosen) [reply Desktop/utilities/imgschemehandler.cpp:19]: same at the other one, the we know it breaks in there. But the thing is, if it does then what?

Comment (shun2wang) [reply Desktop/utilities/imgschemehandler.cpp:19]: Yes, we can do nothing when I find it‘s a Qt bug(maybehttps://bugreports.qt.io/browse/QTBUG-122947). We can only hope for future Qt solutions. Then I will leave it for now.

**Pull request closed without merging**